v-next/core/test/internal/tasks/task-manager.ts

```diff
@@ -0,0 +1,908 @@
+import type { HardhatUserConfig } from "../../../src/types/config.js";
+import type { GlobalParameterMap } from "../../../src/types/global-parameters.js";
+import type { HardhatPlugin } from "../../../src/types/plugins.js";
+
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+
+import { ParameterType } from "../../../src/config.js";
+import { createHardhatRuntimeEnvironment } from "../../../src/index.js";
+import { buildGlobalParameterDefinition } from "../../../src/internal/global-parameters.js";
+import { resolvePluginList } from "../../../src/internal/plugins/resolve-plugin-list.js";
+import {
+  NewTaskDefinitionBuilderImplementation,
+  TaskOverrideDefinitionBuilderImplementation,
+} from "../../../src/internal/tasks/builders.js";
+import { TaskDefinitionType } from "../../../src/types/tasks.js";
+
+/**
+ * There is a circular dependency between the TaskManagerImplementation and the
+ * HardhatRuntimeEnvironmentImplementation. The TaskManagerImplementation needs
+ * the HardhatRuntimeEnvironmentImplementation to be created, and the
+ * HardhatRuntimeEnvironmentImplementation creates the TaskManagerImplementation.
+ *
+ * The way to test the TaskManagerImplementation is through the
+ * HardhatRuntimeEnvironmentImplementation, as it's the one that creates the
+ * TaskManagerImplementation.
+ */
+async function createHre({
+  config = {},
+  plugins = [],
+}: {
+  config?: HardhatUserConfig;
+  plugins?: HardhatPlugin[];
+  globalParameterMap?: GlobalParameterMap;
+} = {}) {
```

Comment:
This function is not needed. Calling `createHardhatRuntimeEnvironment` has the same effect, or should have it. The third parameter shouldn't be used anywhere except for the CLI.